mautic/mautic
# Date time type refactoring

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-380



#### Description:

Resolving this deprecation notice:
```
187x: Using a custom format when the "html5" option of Symfony\Component\Form\Extension\Core\Type\DateTimeType is enabled is deprecated since Symfony 4.3 and will lead to an exception in 5.0.
```

I followed the same approach as described in https://ourcodeworld.com/articles/read/1182/how-to-solve-symfony-5-exception-cannot-use-the-format-option-of-symfony-component-form-extension-core-type-datetype-when-the-html5-option-is-enabled as the situation is the same for us. We have our own JS date picker and don't want to use the built-in HTML5 date picker provided by browsers.

When I was updating the 4th publish up/down field I decided to make the code DRYer and created a trait to define the fields only in 1 place.

This is also fixing a regression introduced in https://github.com/mautic/mautic/pull/11892 where the asset form was throwing 500. We must add some functional tests for assets...

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test that publish up and down fields still work the same. For example in the email, page or focus items edit forms. All date/time fields were affected but all were modified the same way. So if a couple work then the rest will work too.
3. Test also date/time fields for custom fields and form fields.
